Added Metrics Examples Page (#9922)
getsentry/sentry-docs: docs/product/metrics/metrics-examples.mdx

---
title: "Example Use Cases "
sidebar_order: 12
description: "Example Use Cases for custom metrics."
---

If you're not sure where to start, here are a few examples of how you can use custom metrics.

## Frontend Examples
| **Use Case**           | **Metric Type** | **Example** |
| ----------------------------- | --------------------- |
| United States of America (US) | Iowa, USA             |
| European Union (EU)           | Frankfurt, Germany    |

## Backend Examples

> | **Use Case**                                | **Metric Type** | Example|

Author: ```suggestion
| **Use Case**                                | **Metric Type** | Example |
```

---
title: "Example Use Cases"
sidebar_order: 18

> description: "Example Use Cases for custom metrics."

Reviewer: ```suggestion
description: "Learn about some common use cases for using custom metrics."
```

---

> If you're not sure where to start, here are a few examples of how you can use custom metrics.

Reviewer: ```suggestion
You can emit custom metrics to help you track and measure the metrics you care about most. Here are some examples of what you can track with custom metrics. Once your metrics are in Sentry, you'll be able to query, monitor, and get alerted on them. 
```

## Frontend Examples
| **Use Case**                                 | **Metric Type**    | **Example**                                                   |
| -------------------------------------------- | ------------------ | ------------------------------------------------------------- |

> | Count button clicks on subpages              | counter            | Sentry.metrics.increment("button_click", 1, {  tags: { browser: "Firefox", app_version: "1.0.0", subpage: “signup”}}); |

Author: ```suggestion
| Count button clicks on subpages              | counter            | Sentry.metrics.increment("button_click", 1, { tags: { browser: "Firefox", app_version: "1.0.0", subpage: “signup”}}); |
```

> | Number of user signups                       | counter            | Sentry.metrics.increment(”signups”, 1, tags: {plan_type: “freemium”} |

Author: ```suggestion
| Number of user signups                       | counter            | Sentry.metrics.increment(”signups”, 1, tags: {plan_type: “freemium”} ; |
```

> | Number of user signups                       | counter            | Sentry.metrics.increment(”signups”, 1, tags: {plan_type: “freemium”} |

Author: ```suggestion
| Number of user signups                       | counter            | Sentry.metrics.increment(”signups”, 1, { tags: {plan_type: “freemium”}}; |
```

| Measure UX improvement to signup form        | distribution       | Sentry.metrics.distribution("login.time_to_finish_form", 14.3 {tags: { subpage: “signup”}, unit: “second”}); |
| Track the loading times for a component      | distribution       | Sentry.metrics.distribution("component_load_time", 15.0, {tags: { type: "important" },unit: "millisecond",}); |
| Track API request payloads                   | distribution       | Sentry.metrics.distribution("api_request_payload", 136.3, {  tags: { api_name: “maps_provider”, subpage: “signup”}, unit: “kilobyte”}); |
| Count concurrent users in application        | gauge              | Sentry.metrics.gauge("concurrent_users", 10, {  tags: { browser: "Firefox", app_version: "1.0.0"}}); |

> | Track the number of items in a shopping cart | gauge              | Sentry.metrics.gauge("shopping_cart_item_count", itemCount, {tags: { platform: "web" }  |

Author: ```suggestion
| Track the number of items in a shopping cart | gauge              | Sentry.metrics.gauge("shopping_cart_item_count", itemCount, {tags: { platform: "web"}});  |
```

| Track the number of items in a shopping cart       | gauge                   | `Sentry.metrics.gauge("shopping_cart_item_count", itemCount, {`<br />&nbsp;`tags: { platform: "web" }`<br />`});`                                                           |
| Count the unique number of users on a page         | set                     | `Sentry.metrics.set("user_view", "jane", {`<br />&nbsp;`tags: {"page": "/home" }`<br />`});`                                                                                |
| Count unique number of user sessions               | set                     | `Sentry.metrics.set("unique_active_sessions", sessionId, {`<br />&nbsp;`tags: { platform: "mobile" }`<br />`});`                                                            |

## Backend Examples

> | **Use Case**                                       | **Metric Type**    | **Example**                                                                                                                                                                      |

Reviewer: Might be helpful to call out what language these examples are in

| Count of API calls of a 3rd party API              | counter            | sentry_sdk.metrics.incr(key="api_calls", value=1, tags={"api": "third_party"})                                                                        |
| Count of jobs executed per worker                  | counter            | sentry_sdk.metrics.incr(key=”external_api_calls”, value=1,<br> tags={”worker_id”: “485ea324-003d-4f30-a0f9-beed9792c4a1”, “worker_type”: ”standard”}) |

> | Processing time for a task	                       | distribution       | sentry_sdk.metrics.distribution(key=”processing_time”, value=0.002, unit=”second”, tags={”task”: “example_task”})                                     |

Reviewer 2: Could this tab in the first column be the problem?

Author: Unfortunately it wasn't it 😢
